src/libertem/io/dataset/__init__.py

```diff
@@ -99,18 +100,20 @@ def get_dataset_cls(filetype):
     return cls
 
 
-def detect(path, executor):
+def detect(path, executor, getParams=True, getInfo=False):
```

Comment:
Hmm. Seeing this now makes me a bit unhappy - having these boolean parameters is a code smell that this function should really be two functions. I originally suggested the `info` parameter because we could keep backwards-compatibility, but I noticed we don't even document the `detect` method yet. Sorry for flip-flopping on this, but I think it is important to get this function right.

@uellue what do you think about breaking backwards-compat. for `detect`, and documenting the new behavior? I think the only outside user is really the nion swift prototype, and that we can easily PR against to keep it up to date.

So the new version would just behave as if both the `get*` parameters were `True`, and not have those parameters.

(Btw. small style comment: parameters in **Python** usually are `snake_case`, not `camelCase`. I know this can be a bit confusing, as TypeScript is the other way around...)

Reply:
Discussion with @uellue: no need for backward compat. - not documented, only used in prototype code

Reply:
Done! Please do let me know if there are any occurrences of `detect` to be updated and how I should properly document the function. I'm not aware of the 'nion swift' prototype though.

Reply:
For the nion swift integration prototype, see this repository: https://github.com/Brow71189/libertem-hackaton

Reply:
I opened a PR at https://github.com/Brow71189/libertem-hackaton/pull/9.
Are the changes okay?

Reply:
Looks good! Andreas just merged them :tada: